test/index.test.ts

```diff
@@ -594,7 +594,10 @@ describe('asyncWritable', () => {
         (storeValue) => `derived from ${storeValue}`,
         (value, $asyncReadableParent) =>
           Promise.resolve(
-            `constructed from ${value} and ${$asyncReadableParent}`
+            `constructed from ${value} and ${JSON.stringify(
+              $asyncReadableParent
```

Comment:
this is fairly unintuitive as to what is being tested here. I think a new test for this specifically is preferable here.

Reply:
I'll extract it into a new test....